sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -787,6 +790,17 @@ static void _handle_rtr_adv(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
     if (dr == NULL) {
         return;
     }
+#if IS_ACTIVE(MODULE_DHCPV6_CLIENT)
+    if (rtr_adv->flags & NDP_RTR_ADV_FLAGS_M) {
+        if (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA)) {
+            netif->ipv6.aac_mode |= GNRC_NETIF_AAC_DHCP;
+            dhcpv6_client_req_ia_na(netif->pid);
+        }
+        dhcpv6_client_set_conf_mode(DHCPV6_CLIENT_CONF_MODE_STATEFUL);
```

Comment:
How about
```suggestion
    if (rtr_adv->flags & NDP_RTR_ADV_FLAGS_M) {
        if (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA)) {
            netif->ipv6.aac_mode |= GNRC_NETIF_AAC_DHCP;
            dhcpv6_client_req_ia_na(netif->pid);
        }
        else {
            dhcpv6_client_set_conf_mode(DHCPV6_CLIENT_CONF_MODE_STATEFUL);
        }
```

and calling `dhcpv6_client_set_conf_mode()` in `dhcpv6_client_req_ia_na()`. Since requesting `IA_NA` implies stateful mode, I see a potential for config errors, if the latter is not the case.

Reply:
I addressed your suggestion in 46d9a736d8958385b43e2c60078b91b8c3b992ab.